File: backend/src/main/java/com/elbuensabor/entities/MercadoPagoDatos.java
package com.elbuensabor.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;

@Entity
@Table(name = "mercadoPagoDatos")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class MercadoPagoDatos extends Base {
    @Column
    private Long identificadorPago;
    @Column
    private Date fechaCreacion;
    @Column
    private Date fechaAprobacion;
    @Column
    private String formaPago;
    @Column
    private String metodoPago;
    @Column
    private String nroTarjeta;
    @Column
    private String estado;

    @OneToOne
    @JoinColumn(name = "pedido_id")
    private Pedido pedido;
}

File: frontend/el-buen-sabor-frontend/src/app/entidades/MercadoPagoDatos.ts
import { Pedido } from './Pedido';

export class MercadoPagoDatos {
  id: number = 0;
  identificadorPago: number = 0;
  fechaCreacion: Date = new Date();
  fechaAprobacion: Date = new Date();
  formaPago: string = '';
  metodoPago: string = '';
  nroTarjeta: string = '';
  estado: string = '';
  pedido: Pedido = new Pedido();
}
